api/tasks/remove_app_and_related_data_task.py

```diff
@@ -175,30 +177,32 @@ def del_workflow(workflow_id: str):
 
 
 def _delete_app_workflow_runs(tenant_id: str, app_id: str):
-    def del_workflow_run(workflow_run_id: str):
-        db.session.query(WorkflowRun).filter(WorkflowRun.id == workflow_run_id).delete(synchronize_session=False)
-
-    _delete_records(
-        """select id from workflow_runs where tenant_id=:tenant_id and app_id=:app_id limit 1000""",
-        {"tenant_id": tenant_id, "app_id": app_id},
-        del_workflow_run,
-        "workflow run",
+    """Delete all workflow runs for an app using the service repository."""
+    session_maker = sessionmaker(bind=db.engine, expire_on_commit=False)
```

Comment:
The `expire_on_commit=False` argument appears to be unnecessary here since no ORM models have been loaded into the session.